packages/mobile/src/components/block-messages/BlockMessages.tsx

```diff
@@ -0,0 +1,163 @@
+import { cacheUsersSelectors } from '@audius/common'
+import { View } from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
+
+import IconBlockMessages from 'app/assets/images/iconBlockMessages.svg'
+import IconInfo from 'app/assets/images/iconInfo.svg'
+import { Text, Button } from 'app/components/core'
+import { NativeDrawer } from 'app/components/drawer'
+import type { AppState } from 'app/store'
+import { getData } from 'app/store/drawers/selectors'
+import { setVisibility } from 'app/store/drawers/slice'
+import { makeStyles, flexRowCentered } from 'app/styles'
+import { spacing } from 'app/styles/spacing'
+import { useColor } from 'app/utils/theme'
+
+const { getUser } = cacheUsersSelectors
+
+const BLOCK_MESSAGES_MODAL_NAME = 'BlockMessages'
+
+const messages = {
+  title: 'Are you sure?',
+  confirmText1: 'Are you sure you want to block ',
+  confirmText2: ' from sending messages to your inbox?',
+  info: 'This will not affect their ability to view your profile or interact with your content.',
+  blockUser: 'Block User',
+  cancel: 'Cancel'
+}
+
+const useStyles = makeStyles(({ spacing, typography, palette }) => ({
+  drawer: {
```

Comment:
nit: Seems like everything inside the drawer (except the title) is 16px apart - could we we use `gap: spacing(4)` instead of margins on each thing individually? (and then `marginBottom: spacing(2)` on the title container)

Reply:
dude HELL yes